File: sephora-backend/Application/Dtos/Delivery/DeliveryDto.cs
namespace CleanArchitecture.Application.Dtos.Delivery;

public class DeliveryDto
{
    public long Id { get; set; }

    public string Provider { get; set; } = String.Empty;

    public string? PostalCode { get; set; }
    
    public string? Country { get; set; } = "Ukraine";
    
    public string? Region { get; set; } // Область
    
    public string? District { get; set; } // Район
    
    public string? City { get; set; }
    
    public string? Street { get; set; }
    
    public string? Building { get; set; }
    
    public string? Apartment { get; set; }

    public string? NovaPostWarehouse { get; set; }

    public string FirstName { get; set; } = String.Empty;
    
    public string LastName { get; set; } = String.Empty;

    public string PhoneNumber { get; set; } = String.Empty;

    public string Email { get; set; } = String.Empty;
}

File: sephora-frontend/src/models/delivery/DeliveryDto.ts
export default interface DeliveryDto {
    get id(): number;

    get address(): string;

    get provider(): string;

    get firstName(): string;

    get lastName(): string;

    get phoneNumber(): string;

    get email(): string;

    get postalCode(): string | null;

    get region(): string | null;

    get district(): string | null;

    get city(): string | null;

    get street(): string | null;

    get building(): string | null;

    get apartment(): string | null;

    get novaPostWarehouse(): string | null;
}
